routes/post.go

```diff
@@ -1599,17 +1600,601 @@ func (fes *APIServer) GetDiamondedPosts(ww http.ResponseWriter, req *http.Reques
+				requestData.ReaderPublicKeyBase58Check))
+			return
+		}
+	}
+
+	// Get a view with all the mempool transactions.
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetLikesForPost: Error constucting utxoView: %v", err))
+		return
+	}
+
+	// Fetch the likers for the post requested.
+	reclouterPubKeys, err := utxoView.GetLikesForPostHash(postHash)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetLikesForPost: Error getting likers %v", err))
+		return
+	}
+
+	// Filter out any restricted profiles.
+	addReaderPublicKey := false
+	pkMapToFilter := make(map[lib.PkMapKey][]byte)
+	for _, pubKey := range reclouterPubKeys {
+		pkMapKey := lib.MakePkMapKey(pubKey)
+		pkMapToFilter[pkMapKey] = pubKey
+		if reflect.DeepEqual(pubKey, readerPublicKeyBytes) {
+			addReaderPublicKey = true
+		}
+	}
+	var filteredPkMap map[lib.PkMapKey][]byte
+	if addReaderPublicKey {
+		filteredPkMap, err = fes.FilterOutRestrictedPubKeysFromMap(pkMapToFilter, readerPublicKeyBytes, "leaderboard" /*moderationType*/)
+	} else {
+		filteredPkMap, err = fes.FilterOutRestrictedPubKeysFromMap(pkMapToFilter, nil, "leaderboard" /*moderationType*/)
+	}
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetLikesForPost: Error filtering out restricted profiles: %v", err))
+		return
+	}
+
+	// Grab verified username map pointer for constructing profile entry responses.
+	verifiedMap, err := fes.GetVerifiedUsernameToPKIDMap()
+	if err != nil {
+		_AddInternalServerError(ww, fmt.Sprintf("GetLikesForPost: Error fetching verifiedMap: %v", err))
+		return
+	}
+
+	// Create a list of the likers that were not restricted.
+	likers := []*ProfileEntryResponse{}
+	for _, filteredPubKey := range filteredPkMap {
+		profileEntry := utxoView.GetProfileEntryForPublicKey(filteredPubKey)
+		if profileEntry == nil {
+			continue
+		} else {
+			profileEntryResponse := _profileEntryToResponse(profileEntry, fes.Params, verifiedMap, utxoView)
+			likers = append(likers, profileEntryResponse)
+		}
+	}
+
+	// Almost done. Just need to sort the likers.
+	sort.Slice(likers, func(ii, jj int) bool {
+
+		// Attempt to sort on bitclout locked.
+		iiBitCloutLocked := likers[ii].CoinEntry.BitCloutLockedNanos
+		jjBitCloutLocked := likers[jj].CoinEntry.BitCloutLockedNanos
+		if iiBitCloutLocked > jjBitCloutLocked {
+			return true
+		} else if iiBitCloutLocked < jjBitCloutLocked {
+			return false
+		}
+
+		// Sort based on pub key if all else fails.
+		return likers[ii].PublicKeyBase58Check > likers[jj].PublicKeyBase58Check
+	})
+
+	// Cut out the page of reclouters that we care about.
+	likersLength := uint32(len(likers))
+	// Slice the comments from the offset up to either the end of the slice or the offset + limit, whichever is smaller.
+	maxIdx := lib.MinUint32(likersLength, requestData.Offset+requestData.Limit)
+	var likersPage []*ProfileEntryResponse
+	if likersLength > requestData.Offset {
```

Comment:
Okay so I see why you have the if statement written this way, but I would expect to get a slice of length 0 in the response instead of nil. Can we initialize likersPage to an empty slice? this applies to all the `Get{INTERACTION}ForPost` functions